StandaloneMmPkg/Core/StandaloneMmCore.c

```diff
@@ -703,24 +703,41 @@ MigrateMemoryAllocationHobs (
   }
 }
 
-/** Returns the HOB list size.
+/**
+  Check if the MmramRanges don't overlap with all the ranges ranges
+  reported by resource HOB. Also, the HOB list size is returned.
 
-  @param [in]  HobStart   Pointer to the start of the HOB list.
+  @param [in]  HobStart     Pointer to the start of the HOB list.
+  @param [in]  MmramRanges  Pointer to the Mmram ranges.
+  @param [in]  HobStart     Count of Mmram ranges.
 
   @retval Size of the HOB list.
 **/
 UINTN
-GetHobListSize (
-  IN VOID  *HobStart
+CheckResourceHobAndMmramRanges (
+  IN VOID                  *HobStart,
+  IN EFI_MMRAM_DESCRIPTOR  *MmramRanges,
+  IN UINTN                 MmramRangeCount

```

Comment:
GetHobListSize() is a single purpose function which is great.
But the new CheckResoureHobAndMmramRanges() is a mul-purpose function which is not good.
To avoid scanning the hob list twice and also make the code more readable, how about?

Separate a function called InitializeMmHobList (&HobStart, MmramRanges, MmramRangeCount);
InitializeMmHobList is responsible for validating the HOB list and clone to SMRAM if it's valid.
The HobStart is the new location of hob list in SMRAM.

Reply:
Added a new InitializeMmHobList to gather all the operations related to initializing HOB. The the new location of hob list in MMRAM is returned.
VOID *
InitializeMmHobList (
  IN VOID                  *HobStart,
  IN EFI_MMRAM_DESCRIPTOR  *MmramRanges,
  IN UINTN                 MmramRangeCount
  )